### PR TITLE
New 'mismatches' feature on flowcells page (unknown barcodes)

### PR DESCRIPTION
**Pull request by ewels**
For lanes with >= 10 barcodes, the flow cells page now has a column showing the number of mismatches for each barcode for each unknown barcode:

image

If the barcodes are >= 80% matching, the cell is highlighted in red. >= 60% matching cells are highlighted in yellow.

Running on stage: https://genomics-status-stage.scilifelab.se/flowcells/151103_BH5VMNCCXX

### CODE REVIEW

**Comment by Galithil** [comment]
NNNNN is counted as having 0 mismatches, is that on purpose ?

**Comment by Galithil** [comment]
I'm wondering if it makes sense to have colors associated with percentage, rather than with discrete mismatch values, since index length has a small amount of possible values, right ?

**Comment by ewels** [comment]
Yes, it's on purpose - an `N` matches anything, so a full string of `N`s matches all barcodes.

Could go for number of mismatches instead, I don't think it will make much difference though to be honest.

File: run_dir/static/js/flowcell_samples.js [hunk 1]
```diff
@@ -194,7 +202,26 @@ $.getJSON("/api/v1/flowcell_info2/"+flowcell, function(data) {
...
                     count = nice_numbers(count);
-                    ludtable += "<tr"+hl+"><td><samp>"+unmatched+"</samp></td><td>"+count+'</td><td>('+percentage+"%)</span></td></tr>";
+                    ludtable += "<tr"+hl+"><td><samp>"+unmatched+"</samp></td><td>"+count+'</td><td>('+percentage+"%)</span></td>";
+                    if(data['lane'][lid].length <= 10){
+                        for (var s in data['lane'][lid]){
+                            var bc = data['lane'][lid][s]['barcode'];
+                            if(bc !== 'unknown'){
+                                var mm = 0;
+                                for (var i = 0, len = unmatched.length; i < len; i++) {
+                                    if(unmatched[i] != bc[i] && unmatched[i] != 'N' && unmatched[i] != 'N'){
```
Comment (Galithil) [inline run_dir/static/js/flowcell_samples.js]: isnt that the same thing twice ?

Comment (ewels) [inline run_dir/static/js/flowcell_samples.js]: Oops, meant to be `||` not `&&` - so that it matches if either is an `N`

**Comment by Galithil** [comment]
Also, silly question, but if the indexes are of different length, what happens ?

**Comment by ewels** [comment]
It probably crashes :wink:  I don't _think_ that should ever happen though, as the unknown barcodes will always be the same length as those that the demultiplexing was run for, right @vezzi?

**Comment by Galithil** [comment]
@vezzi says that Untedermined will not even be saved with multiple length indexes, so it should be fine.

**Pull request merged**